src/toolbar/toolbar-group.jsx

```diff
@@ -69,87 +130,20 @@ const ToolbarGroup = React.createClass({
     };
   },
 
-  //to update theme inside state whenever a new theme is passed down
-  //from the parent / owner using context
   componentWillReceiveProps(nextProps, nextContext) {
-    const newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
-    this.setState({muiTheme: newMuiTheme});
+    this.setState({
+      muiTheme: nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme,
+    });
   },
 
-  getTheme() {
-    return this.state.muiTheme.toolbar;
+  _handleMouseEnterFontIcon: (style) => (e) => {
```

Comment:
I want to draw your attention here for feed back.

The previous implementation was recomputed the whole style object for each property on mouse enter and leave:
```js
_handleMouseEnterFontIcon(e) {
  e.target.style.zIndex = this.getStyles().icon.hover.zIndex;
e.target.style.color = this.getStyles().icon.hover.color;
},
```

I replaced it with a higher order function so that the style object is only computed once per render. Honestly, if I had my way, we would move these outside of the class too! :grin: 

If these were removed, the only time `this.` is called is for these three scenarios:
1.  `this.props`
2. `this.setState({muiTheme: aMuiTheme})`
3. `this.state.muiTheme`

It's so close to being purely functional!